### PR TITLE
Jetpack Cloud | Add Social upsell when it's not active

### PR DESCRIPTION
**Pull request by manzoorwanijk**
> <details>
> <summary>When neither Jetpack or Social is active, and say Boost is active, users get the following error when they click on enable publicize.</summary>
>
> Image
> 
> To get around it, we need to show a pricing welcome page the same way we do in my Jetpack. 
> 
> 
> 
> Image
>
> </details>

## Proposed Changes

* Show a pricing card when there is no paid plan on the site
* Show the social activation card when there is a paid plan
* Fix the social RNA card image to replace Twitter with Instagram
* Fix spacing in the RNA CTA which had no space between text and external icon

## Why are these changes being made?

- We want to ensure that the user is not left confused when they don't have the Social or Jetpack plugin active
- We want to guide them holding their hand to ensure they are not lost in the error messages that are shown when the plugins are not active

## Testing Instructions

* Ensure that the site does not have a social plan active
* Ensure that neither Social nor Jetpack is active on the site
* Ensure that another standalone plugin like Boost is active
* Goto Jetpack Cloud `/jetpack-social/:site`
* Confirm that you see the pricing card for Social
* Confirm that the Twitter icon on the image is replaced by Instagram
* Click on "Add Jetpack Social"
* Confirm that the checkout is successful
* Now after adding a Social plan to the site, goto `/jetpack-social/:site` again
* Confirm that you see the social activation card instead of the pricing card
* Confirm that the Twitter icon on the image is replaced by Instagram
* Confirm that you see the "Get started" CTA that points to plugin installation/activation page on the site.
* For a non-admin author, confirm that the link points to the plugin page on wp.org.
* Install/Activate the plugin but disable Social sharing
* Reload the `/jetpack-social/:site` page
* Confirm that you now see "Enable Social sharing" as the button label and it points to the plugin settings page.
* Confirm that it points to Social admin page when it's active, otherwise it points to Jetpack settings > Sharing page
* Confirm that everything works same as before when Jetpack or Social is active with publicize enabled
* Confirm that everything is same as before for Simple and Atomic sites

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

- When there is no Social plan on the site
    Screenshot 2024-07-19 at 4 16 38 PM
- For admin user, when there is a paid plan but none of the plugins is active. Notice the CTA URL at the bottom
    Screenshot 2024-07-19 at 4 22 53 PM
- For non-admin user, when there is a paid plan but none of the plugins is active. Notice the CTA URL at the bottom
    Screenshot 2024-07-19 at 4 32 32 PM
- When the plugin is active but sharing is disabled
    Screenshot 2024-07-19 at 4 34 11 PM
- When the plugin is active and sharing is enabled
    Screenshot 2024-07-19 at 4 36 13 PM

### CODE REVIEW

File: client/components/jetpack/card/jetpack-rna-action-card/index.tsx [hunk 1]
```diff
@@ -89,6 +89,7 @@ const JetpackRnaActionCard: React.FC< RnaActionCardProps > = ( {
 						rel={ ctaButtonExternal ? 'noopener noreferrer' : '' }
 					>
 						{ ctaButtonLabel }
+						{ ctaButtonExternal && ' ' }
```
Comment (spsiddarthan) [inline client/components/jetpack/card/jetpack-rna-action-card/index.tsx:92]: Just making sure this was intentional.

**Review by spsiddarthan — COMMENTED** [review]
Thanks for the screenshots. I will give it a test again after this checkout issue is resolved. 

After checkout, it's taking me to a non-existent page on the site. 


https://github.com/user-attachments/assets/df7331e0-974a-4ef8-9ea0-32133b0b4f70

File: client/components/jetpack/card/jetpack-rna-action-card/index.tsx (hunk 1 above)
Comment (manzoorwanijk) [reply client/components/jetpack/card/jetpack-rna-action-card/index.tsx:92]: Yes it is. See the last point in the proposed changes:

> Fix spacing in the RNA CTA which had no space between text and external icon

**Comment by manzoorwanijk** [comment]
> Thanks for the screenshots. I will give it a test again after this checkout issue is resolved.
> 
> After checkout, it's taking me to a non-existent page on the site.
> 

That seems to be a general RNA component or a JN issue. It happens with Scan as well. It works fine for my local site.

**Comment by spsiddarthan** [comment]
I got it into a state where the plugin in active by following the instructions, sharing is enabled, but it still shows this page, and the CTA takes me to the plugin activation page, there is nothing I have to do there since the plugin is active.

Screenshot 2024-07-19 at 6 39 21 PM

**Comment by manzoorwanijk** [comment]
> I got it into a state where the plugin in active by following the instructions, sharing is enabled, but it still shows this page, and the CTA takes me to the plugin activation page, there is nothing I have to do there since the plugin is active.
> 

May be some sync/cache issue. Can you please try reloading the page after ensuring that the site/system is not sandboxed?

**Comment by spsiddarthan** [comment]
>May be some sync/cache issue

Possible it's the former. I had to refresh three times for Jetpack Cloud to know that the plugin is active, and that's fine for now. Let's merge this and see if it happens on prod, and if it does, we can think of ways to improve it.

**Review by spsiddarthan — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16407728

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @manzoorwanijk for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16407728).

**Pull request merged**